test/unit/util/locking/RedisResourceLocker.test.ts

```diff
@@ -0,0 +1,169 @@
+        return lock;
+      },
+    );
+  });
+
+  it('can lock and unlock a resource.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(1);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+  });
+
+  it('can lock a resource again after it was unlocked.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(2);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+  });
+
+  it('errors when unlocking a resource that was not locked.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).rejects.toThrow(InternalServerError);
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(1);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+  });
+
+  it('errors when redlock.lock throws an error.', async(): Promise<void> => {
+    locker.redlock.lock = jest.fn().mockRejectedValue(new Error('random Error'));
+    await expect(locker.acquire(identifier)).rejects.toThrow(InternalServerError);
+    await expect(locker.acquire(identifier)).rejects.toThrow('Unable to acquire lock for ');
+    await expect(locker.acquire(identifier)).rejects.toThrow('Error: random Error');
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(3);
+  });
+
+  it('errors when redlock.unlock throws an error.', async(): Promise<void> => {
+    await locker.acquire(identifier);
+    locker.redlock.unlock = jest.fn().mockRejectedValue(new Error('random Error'));
+    await expect(locker.release(identifier)).rejects.toThrow(InternalServerError);
+    await expect(locker.release(identifier)).rejects.toThrow('Unable to release lock for: ');
+    await expect(locker.release(identifier)).rejects.toThrow('Error: random Error');
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(3);
+    locker.redlock.unlock = jest.fn();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+  });
+
+  it('does not extend when lockList does not contain the identifiers lock to extend.', async(): Promise<void> => {
+    jest.useFakeTimers();
+    await locker.acquire(identifier);
+    locker.lockList = new Map();
+    jest.advanceTimersByTime(20000);
+    expect(locker.redlock.extend).toHaveBeenCalledTimes(0);
+  });
+
+  it('when redlock.extend throws an error, delete the timeout for that lock and' +
+      'make sure extend is only called once.', async(): Promise<void> => {
```

Comment:
Would rephrase so it fits on a single line (currently it would also have the word `andmake` since there is no space there).